src/azureFunction/azureFunctionProjectService.ts

```diff
@@ -0,0 +1,186 @@
+		const newFilePromise = getNewFunctionFile(projectFile);
+
+		// get function name from user
+		const functionName = await vscode.window.showInputBox({
+			title: constants.functionNameTitle,
+			value: table,
+			ignoreFocusOut: true
+		});
+		if (!functionName) {
+			return;
+		}
+
+		// create C# HttpTrigger
+		await azureFunctionApi.createFunction({
+			language: 'C#',
+			templateId: 'HttpTrigger',
+			functionName: functionName,
+			folderPath: projectFile
+		});
+
+		// once steps are addressed we add SQL Bindings Nuget and Connection String to local.settings.json
+		// and then create a new functions file
+		await addNugetReferenceToProjectFile(projectFile);
+		await addConnectionStringToConfig(connectionString);
+		const functionFile = await newFilePromise;
+
+		let objectName = generateQuotedFullName(schema, table);
+		await this.azureFunctionsService.addSqlBinding(
+			mssql.BindingType.input,
+			functionFile,
+			functionName,
+			objectName,
+			constants.sqlConnectionString
+		);
+
+		this.overwriteAzureFunctionMethodBody(functionFile);
+	}
+
+	/**
+	 * Overwrites the Azure function file to include the new binding in the method body
+	 * @param filePath is the path for the function file (.cs for C# functions)
+	 */
+	private overwriteAzureFunctionMethodBody(filePath: string): void {
+		let defaultBindedFunctionText = fs.readFileSync(filePath, 'utf-8');
+		// Replace default binding text
+		let newValueLines = defaultBindedFunctionText.split(os.EOL);
+		const defaultLineSet = new Set(defaultSqlBindingTextLines);
+		let replacedValueLines = [];
+		for (let defaultLine of newValueLines) {
+			// Skipped lines
+			if (defaultLineSet.has(defaultLine.trimStart())) {
+				continue;
+			} else if (defaultLine.trimStart() === defaultBindingResult) { // Result change
+				replacedValueLines.push(defaultLine.replace(defaultBindingResult, sqlBindingResult));
+			} else {
+				// Normal lines to be included
+				replacedValueLines.push(defaultLine);
+			}
+		}
+		defaultBindedFunctionText = replacedValueLines.join(os.EOL);
+		fs.writeFileSync(filePath, defaultBindedFunctionText, 'utf-8');
+	}
+}
+
+/**
+ * A C# Azure Functions project must be present in order to create a new Azure Function for the table
+ * @returns the selected project file path or indicates to user there is no C# Azure Functions project
+ */
+async function isAzureFunctionProjectOpen(): Promise<string> {
+	let selectedProjectFile: string | undefined = '';
+	if (vscode.workspace.workspaceFolders === undefined || vscode.workspace.workspaceFolders.length === 0) {
+		return selectedProjectFile;
+	} else {
+		const hostFiles = await getHostFile();
+		const projectFiles = await getProjectFile();
+		if (projectFiles !== undefined && hostFiles !== undefined) {
+			// Check to see if its only a function project using host.json and project file
+			let functionProjects = [];
+			let count = 0;
+			for (let host of hostFiles) {
+				for (let project of projectFiles) {
```

Comment:
Easier and more readable to just do 

`const functionsProjects = projectFiles.filter(p => ...)`